sherpa/plot/backends.py

```diff
@@ -0,0 +1,538 @@
+
+        Returns
+        -------
+        latex : str
+            The txt modified as appropriate for a backend so that the LaTeX
+            will be displayed properly.
+
+        """
+        return txt
+
+    # HTML representation as tabular data
+    #
+    def as_html(self, data, fields):
+        """Create HTML representation of a plot
+
+        Parameters
+        ----------
+        data : Plot instance
+            The plot object to display.
+        fields : sequence of strings
+            The fields of data to use.
+
+        """
+
+        # Would like a nicer way to set the summary label, but without
+        # adding a per-class field for this it is safest just to use
+        # the object name.
+
+        meta = []
+        for name in fields:
+            # skip records which we don't know about. This indicates
+            # an error in the calling code, but we don't want it to
+            # stop the generation of the HTML.
+            #
+            try:
+                val = getattr(data, name)
+            except Exception as e:
+                lgr.debug("Skipping field {}: {}".format(name, e))
+                continue
+
+            meta.append((name, val))
+
+        ls = [formatting.html_section(meta, open_block=True,
+                                    summary=type(data).__name__)]
+        return formatting.html_from_sections(data, ls)
+
+    # The follwowing methods will almost all be removed in Step 2
+    def get_split_plot_defaults(self):
+        return get_keyword_defaults(self.set_subplot, 3)
+
+
+    def get_plot_defaults(self):
+        return get_keyword_defaults(self.plot, 7)
+
+
+    def get_point_defaults(self):
+        return get_keyword_defaults(self.point, 2)
+
+
+    def get_contour_defaults(self):
+        return get_keyword_defaults(self.contour, 6)
+
+
+    def get_histo_defaults(self):
+        return get_keyword_defaults(self.histo, 6)
+
+
+    def get_dummy_defaults(self):
+        return {}
+
+
+    get_data_plot_defaults = get_dummy_defaults
+    get_model_plot_defaults = get_dummy_defaults
+    get_fit_plot_defaults = get_dummy_defaults
+    get_resid_plot_defaults = get_dummy_defaults
+    get_ratio_plot_defaults = get_dummy_defaults
+
+    get_data_contour_defaults = get_dummy_defaults
+    get_model_contour_defaults = get_dummy_defaults
+    get_fit_contour_defaults = get_dummy_defaults
+    get_resid_contour_defaults = get_dummy_defaults
+    get_ratio_contour_defaults = get_dummy_defaults
+
+    get_confid_point_defaults = get_dummy_defaults
+    get_confid_plot_defaults = get_dummy_defaults
+    get_confid_contour_defaults = get_dummy_defaults
+    get_model_histo_defaults = get_dummy_defaults
+    get_component_plot_defaults = get_dummy_defaults
+    get_component_histo_defaults = get_dummy_defaults
+    get_scatter_plot_defaults = get_dummy_defaults
+    get_cdf_plot_defaults = get_dummy_defaults
+
+    def as_html_histogram(self, plot):
```

Comment:
I introduced these "as htmll" style functions to support the notebook code, but always thought that they felt "ugly". At least here it is "in a better place"!

Reply:
It's still not a good place - I discussed that in #1375: In a later step, I think that any part that is specific to the plot type (DataPlot, ModelHistrogramPlot, ComplextNotYetInventedPlot) should live in the plot class and then we have only a single `anything_as_html` in the backend and can add new plot ypes without changing the backends, but that's for the next PR. (I'm answering as you go, in case we want to have a 2-way conversation on some of these issues).

Reply:
Looking at the documentation you see this whole set of `as_html_xxx` routines and whilst this makes it obvious/easy to check if a particular "plot type" is supported, you do wonder if maybe we should have


    as_html(self, data, fields)
    as_html(sellf, plottype, data)

with the latter for things like `p.as_html("histogram", histodata)`, but then the two versions are harder to separate (there's actually an easy way of "is the first element a string we care about?" but its not ideal). It would slightly reduce the size of the objects (and a lot of repeated documentation) but is it a worthwhile change? I guess there could be a separate routine for the latter (rather than having n of them). But it's the sort of thing I wonder why I didn't mention before.